modules/search-service-impl/src/main/java/org/opencastproject/search/impl/solr/SolrIndexManager.java

```diff
@@ -981,12 +987,13 @@ static void addMpeg7Metadata(SolrInputDocument doc, MediaPackage mediaPackage, M
       sortedAnnotations = new TreeSet<TextAnnotation>(new Comparator<TextAnnotation>() {
         @Override
         public int compare(TextAnnotation a1, TextAnnotation a2) {
-          if ((RELEVANCE_BOOST * a1.getRelevance() + a1.getConfidence()) > (RELEVANCE_BOOST * a2.getRelevance() + a2
-                  .getConfidence()))
+          double v1 = RELEVANCE_BOOST * a1.getRelevance() + a1.getConfidence();
+          double v2 = RELEVANCE_BOOST * a2.getRelevance() + a2.getConfidence();
+          if (v1 > v2) {
             return -1;
-          else if ((RELEVANCE_BOOST * a1.getRelevance() + a1.getConfidence()) < (RELEVANCE_BOOST * a2.getRelevance() + a2
-                  .getConfidence()))
+          } else if (v1 < v2) {
             return 1;
+          }
```

Comment:
Unlike the rest of this PR (and previous PRs), this change is actually not super boring. So you might want to actually review this piece of code to make sure my change does not alter the behavior of the program.